python/dgl/transforms/functional.py

```diff
@@ -3300,4 +3302,69 @@ def norm_by_dst(g, etype=None):
 
     return norm
 
+def rwpe(g, k):
+    r"""Return random walk positional encodings.
+
+    For internal use.
+
+    Parameters
+    ----------
+    g : DGLGraph
+        The homogeneous graph.
+    k : int
+        The number of random walk steps.
+
+    Returns
+    -------
+    Tensor
+        The random walk positional encodings.
+    """
+    # get 1-step random walk probabilities as A * D^-1
+    A = g.adj(scipy_fmt='csr') # adjacency matrix
+    Dinv = sparse.diags(F.asnumpy(g.in_degrees()).clip(1) ** -1.0, dtype=float) # D^-1
+    RW = A * Dinv
```

Comment:
Should we change "A * Dinv" to "Dinv * A"?

Reply:
Both are fine because we only use the diagonal elements. The paper actually uses the expression 'A * Dinv'.